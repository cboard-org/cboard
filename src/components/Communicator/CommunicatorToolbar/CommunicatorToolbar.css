.CommunicatorToolbar {
  position: relative;
  display: flex;
  justify-content: space-around;
  height: 40px;
  color: #fff;
  background-color: #4b4b4b;
  white-space: nowrap;
  align-items: center;
}

.CommunicatorToolbar button {
  color: #fff;
}
  .CommunicatorToolbar button icon {
    margin-left: 0;
  }

.CommunicatorToolbar__menu {
  text-transform: capitalize;
}

.CommunicatorToolbar__group {
  display: flex;
  flex: 1 1;
  align-items: center;
}

.CommunicatorToolbar .Communicator__title {
  display: flex;
  order: 2;
  align-items: center;
  justify-content: center;
  top: 0;
  right: 0;
  height: 100%;
  margin: 0;
  font-size: 1.14rem;
  font-weight: 400;
  text-transform: uppercase;
}

.Communicator__title.logged-in {
  cursor: pointer;
}

.CommunicatorToolbar__group--start {
  justify-content: flex-start;
}

.CommunicatorToolbar__group--end {
  order: 2;
  justify-content: flex-end;
}

.CommunicatorToolbar__menuitem {
  max-width: 360px;
}
