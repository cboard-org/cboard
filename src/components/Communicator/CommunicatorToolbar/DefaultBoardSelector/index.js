import DefaultBoardSelector from './DefaultBoardSelector';

export default DefaultBoardSelector;
