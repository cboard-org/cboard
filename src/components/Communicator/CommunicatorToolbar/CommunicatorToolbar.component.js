import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, intlShape } from 'react-intl';
import classNames from 'classnames';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import ListItem from '@material-ui/core/ListItem';
import Menu from '@material-ui/core/Menu';
import ListItemText from '@material-ui/core/ListItemText';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import Avatar from '@material-ui/core/Avatar';
import LayersIcon from '@material-ui/icons/Layers';
import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
import ViewModuleIcon from '@material-ui/icons/ViewModule';
import AddCircleIcon from '@material-ui/icons/AddCircle';

import FormDialog from '../../UI/FormDialog';
import messages from './CommunicatorToolbar.messages';
import './CommunicatorToolbar.css';
import { isCordova } from '../../../cordova-util';
import DefaultBoardSelector from './DefaultBoardSelector';

class CommunicatorToolbar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      boardsMenu: null,
      openTitleDialog: false,
      titleDialogValue:
        this.props.currentCommunicator.name ||
        this.props.currentCommunicator.id ||
        ''
    };
  }

  openMenu(e) {
    this.setState({ boardsMenu: e.currentTarget });
  }

  closeMenu() {
    this.setState({ boardsMenu: null });
  }

  switchBoard(board) {
    this.closeMenu();
    this.props.switchBoard(board.id);
    this.props.history.replace(`/board/${board.id}`);
  }

  handleCommunicatorTitleClick = () => {
    if (!this.props.isLoggedIn) {
      return false;
    }

    if (!this.isCommunicatorTitleClicked) {
      this.isCommunicatorTitleClicked = setTimeout(() => {
        this.isCommunicatorTitleClicked = false;
      }, 400);
    } else {
      this.setState({
        openTitleDialog: true,
        titleDialogValue:
          this.props.currentCommunicator.name ||
          this.props.currentCommunicator.id ||
          ''
      });
    }
  };

  handleCommunicatorTitleChange = event => {
    const { value: titleDialogValue } = event.target;
    this.setState({ titleDialogValue });
  };

  handleCommunicatorTitleSubmit = async () => {
    if (this.state.titleDialogValue.length) {
      try {
        await this.props.editCommunicatorTitle(this.state.titleDialogValue);
      } catch (e) {}
    }
    this.handleCommunicatorTitleClose();
  };

  handleCommunicatorTitleClose = () => {
    this.setState({
      openTitleDialog: false,
      titleDialogValue:
        this.props.currentCommunicator.name ||
        this.props.currentCommunicator.id ||
        ''
    });
  };

  handleNewBoardClick = () => {};

  boardCaption = board => {
    // Cordova path cannot be absolute
    if (isCordova() && board.caption && board.caption.search('/') === 0) {
      return `.${board.caption}`;
    } else {
      return board.caption;
    }
  };

  render() {
    const {
      intl,
      className,
      boards,
      isSelecting,
      openCommunicatorDialog,
      isDark,
      changeDefaultBoard
    } = this.props;

    return (
      <div className={classNames('CommunicatorToolbar', className)}>
        <Button
          className="Communicator__title"
          id="boards-button"
          disabled={isSelecting || boards.length === 0}
          onClick={this.openMenu.bind(this)}
        >
          <ArrowDropDownIcon />
          <FormattedMessage {...messages.boards} />
        </Button>
        <Menu
          id="boards-menu"
          className="CommunicatorToolbar__menu"
          anchorEl={this.state.boardsMenu}
          open={Boolean(this.state.boardsMenu)}
          onClose={this.closeMenu.bind(this)}
        >
          {boards.map(board => (
            <ListItem
              className="CommunicatorToolbar__menuitem"
              key={board.id}
              onClick={this.switchBoard.bind(this, board)}
            >
              <ListItemAvatar>
                {this.boardCaption(board) ? (
                  <Avatar src={this.boardCaption(board)} />
                ) : (
                  <Avatar>
                    <ViewModuleIcon />
                  </Avatar>
                )}
              </ListItemAvatar>
              <ListItemText
                inset
                primary={
                  board.name ||
                  (board.nameKey &&
                    intl.formatMessage({
                      id: board.nameKey,
                      defaultMessage: board.id
                    }))
                }
                secondary={intl.formatMessage(messages.tiles, {
                  qty: board.tiles.length
                })}
              />
            </ListItem>
          ))}
        </Menu>
        <FormDialog
          open={this.state.openTitleDialog}
          title={<FormattedMessage {...messages.editTitle} />}
          onSubmit={this.handleCommunicatorTitleSubmit}
          onClose={this.handleCommunicatorTitleClose}
        >
          <TextField
            autoFocus
            margin="dense"
            label={<FormattedMessage {...messages.communicatorTitle} />}
            value={this.state.titleDialogValue}
            type="text"
            onChange={this.handleCommunicatorTitleChange}
            fullWidth
            required
          />
        </FormDialog>

        <div className="CommunicatorToolbar__group CommunicatorToolbar__group--start">
          <Button
            className={'edit__communicator'}
            disabled={isSelecting}
            onClick={openCommunicatorDialog}
          >
            <LayersIcon className="CommunicatorToolbar__group CommunicatorToolbar__group--start--button" />
            {intl.formatMessage(messages.editCommunicator)}
          </Button>
        </div>
        <div className="CommunicatorToolbar__group CommunicatorToolbar__group--end">
          {false && (
            <div>
              <Button
                label={intl.formatMessage(messages.addBoardButton)}
                onClick={this.handleNewBoardClick}
                //TODO: need to implement function
                disabled={isSelecting}
                color="inherit"
              >
                {intl.formatMessage(messages.addBoardButton)}
                <AddCircleIcon />
              </Button>
            </div>
          )}
          <DefaultBoardSelector
            isDarkMode={isDark}
            changeDefaultBoard={changeDefaultBoard}
          />
        </div>
      </div>
    );
  }
}

CommunicatorToolbar.defaultProps = {
  className: '',
  boards: [],
  isSelecting: false,
  switchBoard: () => {},
  showNotification: () => {},
  openCommunicatorDialog: () => {},
  changeDefaultBoard: () => {}
};

CommunicatorToolbar.propTypes = {
  className: PropTypes.string,
  intl: intlShape.isRequired,
  boards: PropTypes.array,
  currentCommunicator: PropTypes.object,
  isSelecting: PropTypes.bool,
  showNotification: PropTypes.func,
  switchBoard: PropTypes.func,
  openCommunicatorDialog: PropTypes.func,
  editCommunicatorTitle: PropTypes.func,
  isDark: PropTypes.bool,
  changeDefaultBoard: PropTypes.func
};

export default CommunicatorToolbar;
