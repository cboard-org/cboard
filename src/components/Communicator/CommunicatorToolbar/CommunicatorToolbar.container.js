import React from 'react';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import CommunicatorToolbar from './CommunicatorToolbar.component';
import CommunicatorDialog from '../CommunicatorDialog';
import {
  switchBoard,
  replaceBoard,
  changeDefaultBoard
} from '../../Board/Board.actions';
import { showNotification } from '../../Notifications/Notifications.actions';
import {
  importCommunicator,
  createCommunicator,
  deleteCommunicator,
  changeCommunicator,
  upsertCommunicator
} from '../Communicator.actions';
import API from '../../../api';

class CommunicatorContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      openDialog: false
    };
  }

  openCommunicatorDialog() {
    this.setState({ openDialog: true });
  }

  closeCommunicatorDialog() {
    this.setState({ openDialog: false });
  }

  editCommunicatorTitle = async name => {
    const updatedCommunicatorData = {
      ...this.props.currentCommunicator,
      name
    };

    const communicatorData = await API.updateCommunicator(
      updatedCommunicatorData
    );

    this.props.upsertCommunicator(communicatorData);
    this.props.changeCommunicator(communicatorData.id);
  };

  render() {
    const toolbarProps = {
      ...this.props,
      isLoggedIn: !!this.props.userData.email,
      editCommunicatorTitle: this.editCommunicatorTitle,
      openCommunicatorDialog: this.openCommunicatorDialog.bind(this)
    };

    return (
      <React.Fragment>
        <CommunicatorToolbar {...toolbarProps} />
        {this.state.openDialog && (
          <CommunicatorDialog
            open={this.state.openDialog}
            onClose={this.closeCommunicatorDialog.bind(this)}
          />
        )}
      </React.Fragment>
    );
  }
}

const mapStateToProps = (
  { board, communicator, app: { userData, displaySettings } },
  ownProps
) => {
  const activeCommunicatorId = communicator.activeCommunicatorId;
  const currentCommunicator = communicator.communicators.find(
    communicator => communicator.id === activeCommunicatorId
  );
  const activeBoardId = board.activeBoardId;
  const boards = board.boards.filter(
    board =>
      board !== null &&
      board.id !== null &&
      currentCommunicator.boards.includes(board.id)
  );
  const currentBoard = boards.find(board => board.id === activeBoardId);

  return {
    communicators: communicator.communicators,
    boards,
    currentCommunicator,
    currentBoard,
    userData,
    isDark: displaySettings.darkThemeActive,
    ...ownProps
  };
};

const mapDispatchToProps = {
  importCommunicator,
  createCommunicator,
  deleteCommunicator,
  changeCommunicator,
  upsertCommunicator,
  showNotification,
  switchBoard,
  replaceBoard,
  changeDefaultBoard
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(withRouter(CommunicatorContainer)));
