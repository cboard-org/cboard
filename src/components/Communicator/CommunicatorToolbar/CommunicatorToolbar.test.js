import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommunicatorToolbar from './CommunicatorToolbar.component';

jest.mock('./CommunicatorToolbar.messages', () => {
  return {
    communicators: {
      id: 'cboard.components.CommunicatorToolbar.communicators',
      defaultMessage: 'Communicators'
    },
    editTitle: {
      id: 'cboard.components.CommunicatorToolbar.editTitle',
      defaultMessage: 'Edit Communicator Title'
    },
    communicatorTitle: {
      id: 'cboard.components.CommunicatorToolbar.communicatorTitle',
      defaultMessage: 'Communicator Title'
    },
    boards: {
      id: 'cboard.components.CommunicatorToolbar.boards',
      defaultMessage: 'Boards'
    },
    copyMessage: {
      id: 'cboard.components.CommunicatorToolbar.copyMessage',
      defaultMessage: 'Copied to clipboard!'
    },
    share: {
      id: 'cboard.components.CommunicatorToolbar.share',
      defaultMessage: 'Share'
    },
    tiles: {
      id: 'cboard.components.CommunicatorToolbar.tiles',
      defaultMessage: 'Tiles'
    },
    editCommunicator: {
      id: 'cboard.components.CommunicatorToolbar.editCommunicator',
      defaultMessage: 'Build'
    },
    addBoardButton: {
      id: 'cboard.components.CommunicatorToolbar.addBoardButton',
      defaultMessage: 'Add Board'
    }
  };
});

let switchBoardsCount = 0;
const intlMock = {
  formatMessage: ({ id }) => id
};

const COMPONENT_PROPS = {
  intl: intlMock,
  isSelecting: false,
  boards: [
    {
      id: 'board-1',
      nameKey: 'board-1-name-key',
      tiles: []
    },
    {
      id: 'board-2',
      name: 'board-2-name',
      tiles: []
    }
  ],
  switchBoard: () => {
    switchBoardsCount++;
  },
  currentCommunicator: {
    name: 'some-communicator-title'
  },
  history: {
    replace: () => {}
  },
  openCommunicatorDialog: () => {},
  editCommunicatorTitle: () => {}
};

describe('Communicator tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<CommunicatorToolbar {...COMPONENT_PROPS} />);
  });

  test('menu behavior', () => {
    const wrapper = shallow(<CommunicatorToolbar {...COMPONENT_PROPS} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    let boardsMenu = wrapper.find('#boards-menu').get(0);
    expect(boardsMenu.props.open).toBe(false);

    const boardsButton = wrapper.find('#boards-button');
    boardsButton.simulate('click', { currentTarget: 'someElement' });

    boardsMenu = wrapper.find('#boards-menu').get(0);
    expect(boardsMenu.props.open).toBe(true);

    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  test('switching boards behavior', () => {
    const wrapper = shallow(<CommunicatorToolbar {...COMPONENT_PROPS} />);
    let tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    let boardsMenu = wrapper.find('#boards-menu').get(0);
    expect(boardsMenu.props.open).toBe(false);

    const boardsButton = wrapper.find('#boards-button');
    boardsButton.simulate('click', { currentTarget: 'someElement' });

    boardsMenu = wrapper.find('#boards-menu').get(0);
    expect(boardsMenu.props.open).toBe(true);

    tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

    expect(switchBoardsCount).toBe(0);

    const firstBoardOption = wrapper.find('WithStyles(ListItem)').get(0);
    firstBoardOption.props.onClick();
    expect(switchBoardsCount).toBe(1);

    const secondBoardOption = wrapper.find('WithStyles(ListItem)').get(1);
    secondBoardOption.props.onClick();
    expect(switchBoardsCount).toBe(2);
  });
});
