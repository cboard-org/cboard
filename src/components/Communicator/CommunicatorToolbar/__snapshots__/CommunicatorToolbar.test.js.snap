// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Communicator tests default renderer 1`] = `
<div
  className="CommunicatorToolbar"
>
  <ForwardRef(WithStyles(ForwardRef(Button)))
    className="Communicator__title"
    disabled={false}
    id="boards-button"
    onClick={[Function]}
  >
    <UNDEFINED />
    <FormattedMessage
      defaultMessage="Boards"
      id="cboard.components.CommunicatorToolbar.boards"
      values={Object {}}
    />
  </ForwardRef(WithStyles(ForwardRef(Button)))>
  <ForwardRef(WithStyles(ForwardRef(Menu)))
    anchorEl={null}
    className="CommunicatorToolbar__menu"
    id="boards-menu"
    onClose={[Function]}
    open={false}
  >
    <ForwardRef(WithStyles(ForwardRef(ListItem)))
      className="CommunicatorToolbar__menuitem"
      onClick={[Function]}
    >
      <ForwardRef(WithStyles(ForwardRef(ListItemAvatar)))>
        <ForwardRef(WithStyles(ForwardRef(Avatar)))>
          <UNDEFINED />
        </ForwardRef(WithStyles(ForwardRef(Avatar)))>
      </ForwardRef(WithStyles(ForwardRef(ListItemAvatar)))>
      <ForwardRef(WithStyles(ForwardRef(ListItemText)))
        inset={true}
        primary="board-1-name-key"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </ForwardRef(WithStyles(ForwardRef(ListItem)))>
    <ForwardRef(WithStyles(ForwardRef(ListItem)))
      className="CommunicatorToolbar__menuitem"
      onClick={[Function]}
    >
      <ForwardRef(WithStyles(ForwardRef(ListItemAvatar)))>
        <ForwardRef(WithStyles(ForwardRef(Avatar)))>
          <UNDEFINED />
        </ForwardRef(WithStyles(ForwardRef(Avatar)))>
      </ForwardRef(WithStyles(ForwardRef(ListItemAvatar)))>
      <ForwardRef(WithStyles(ForwardRef(ListItemText)))
        inset={true}
        primary="board-2-name"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </ForwardRef(WithStyles(ForwardRef(ListItem)))>
  </ForwardRef(WithStyles(ForwardRef(Menu)))>
  <WithWidth(WithMobileDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title={
      <FormattedMessage
        defaultMessage="Edit Communicator Title"
        id="cboard.components.CommunicatorToolbar.editTitle"
        values={Object {}}
      />
    }
  >
    <ForwardRef(WithStyles(ForwardRef(TextField)))
      autoFocus={true}
      fullWidth={true}
      label={
        <FormattedMessage
          defaultMessage="Communicator Title"
          id="cboard.components.CommunicatorToolbar.communicatorTitle"
          values={Object {}}
        />
      }
      margin="dense"
      onChange={[Function]}
      required={true}
      type="text"
      value="some-communicator-title"
    />
  </WithWidth(WithMobileDialog)>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--start"
  >
    <ForwardRef(WithStyles(ForwardRef(Button)))
      className="edit__communicator"
      disabled={false}
      onClick={[Function]}
    >
      <UNDEFINED
        className="CommunicatorToolbar__group CommunicatorToolbar__group--start--button"
      />
      cboard.components.CommunicatorToolbar.editCommunicator
    </ForwardRef(WithStyles(ForwardRef(Button)))>
  </div>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--end"
  >
    <DefaultBoardSelector
      changeDefaultBoard={[Function]}
    />
  </div>
</div>
`;

exports[`Communicator tests menu behavior 1`] = `
<div
  className="CommunicatorToolbar"
>
  <WithStyles(ForwardRef(Button))
    className="Communicator__title"
    disabled={false}
    id="boards-button"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ArrowDropDownIcon)) />
    <FormattedMessage
      defaultMessage="Boards"
      id="cboard.components.CommunicatorToolbar.boards"
      values={Object {}}
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    className="CommunicatorToolbar__menu"
    id="boards-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-1"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-1-name-key"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-2"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-2-name"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(Menu))>
  <WithWidth(WithMobileDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title={
      <FormattedMessage
        defaultMessage="Edit Communicator Title"
        id="cboard.components.CommunicatorToolbar.editTitle"
        values={Object {}}
      />
    }
  >
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      fullWidth={true}
      label={
        <FormattedMessage
          defaultMessage="Communicator Title"
          id="cboard.components.CommunicatorToolbar.communicatorTitle"
          values={Object {}}
        />
      }
      margin="dense"
      onChange={[Function]}
      required={true}
      type="text"
      value="some-communicator-title"
    />
  </WithWidth(WithMobileDialog)>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--start"
  >
    <WithStyles(ForwardRef(Button))
      className="edit__communicator"
      disabled={false}
      onClick={[Function]}
    >
      <Memo(ForwardRef(LayersIcon))
        className="CommunicatorToolbar__group CommunicatorToolbar__group--start--button"
      />
      cboard.components.CommunicatorToolbar.editCommunicator
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--end"
  >
    <DefaultBoardSelector
      changeDefaultBoard={[Function]}
    />
  </div>
</div>
`;

exports[`Communicator tests menu behavior 2`] = `
<div
  className="CommunicatorToolbar"
>
  <WithStyles(ForwardRef(Button))
    className="Communicator__title"
    disabled={false}
    id="boards-button"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ArrowDropDownIcon)) />
    <FormattedMessage
      defaultMessage="Boards"
      id="cboard.components.CommunicatorToolbar.boards"
      values={Object {}}
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    anchorEl="someElement"
    className="CommunicatorToolbar__menu"
    id="boards-menu"
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-1"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-1-name-key"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-2"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-2-name"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(Menu))>
  <WithWidth(WithMobileDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title={
      <FormattedMessage
        defaultMessage="Edit Communicator Title"
        id="cboard.components.CommunicatorToolbar.editTitle"
        values={Object {}}
      />
    }
  >
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      fullWidth={true}
      label={
        <FormattedMessage
          defaultMessage="Communicator Title"
          id="cboard.components.CommunicatorToolbar.communicatorTitle"
          values={Object {}}
        />
      }
      margin="dense"
      onChange={[Function]}
      required={true}
      type="text"
      value="some-communicator-title"
    />
  </WithWidth(WithMobileDialog)>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--start"
  >
    <WithStyles(ForwardRef(Button))
      className="edit__communicator"
      disabled={false}
      onClick={[Function]}
    >
      <Memo(ForwardRef(LayersIcon))
        className="CommunicatorToolbar__group CommunicatorToolbar__group--start--button"
      />
      cboard.components.CommunicatorToolbar.editCommunicator
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--end"
  >
    <DefaultBoardSelector
      changeDefaultBoard={[Function]}
    />
  </div>
</div>
`;

exports[`Communicator tests switching boards behavior 1`] = `
<div
  className="CommunicatorToolbar"
>
  <WithStyles(ForwardRef(Button))
    className="Communicator__title"
    disabled={false}
    id="boards-button"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ArrowDropDownIcon)) />
    <FormattedMessage
      defaultMessage="Boards"
      id="cboard.components.CommunicatorToolbar.boards"
      values={Object {}}
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    anchorEl={null}
    className="CommunicatorToolbar__menu"
    id="boards-menu"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-1"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-1-name-key"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-2"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-2-name"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(Menu))>
  <WithWidth(WithMobileDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title={
      <FormattedMessage
        defaultMessage="Edit Communicator Title"
        id="cboard.components.CommunicatorToolbar.editTitle"
        values={Object {}}
      />
    }
  >
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      fullWidth={true}
      label={
        <FormattedMessage
          defaultMessage="Communicator Title"
          id="cboard.components.CommunicatorToolbar.communicatorTitle"
          values={Object {}}
        />
      }
      margin="dense"
      onChange={[Function]}
      required={true}
      type="text"
      value="some-communicator-title"
    />
  </WithWidth(WithMobileDialog)>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--start"
  >
    <WithStyles(ForwardRef(Button))
      className="edit__communicator"
      disabled={false}
      onClick={[Function]}
    >
      <Memo(ForwardRef(LayersIcon))
        className="CommunicatorToolbar__group CommunicatorToolbar__group--start--button"
      />
      cboard.components.CommunicatorToolbar.editCommunicator
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--end"
  >
    <DefaultBoardSelector
      changeDefaultBoard={[Function]}
    />
  </div>
</div>
`;

exports[`Communicator tests switching boards behavior 2`] = `
<div
  className="CommunicatorToolbar"
>
  <WithStyles(ForwardRef(Button))
    className="Communicator__title"
    disabled={false}
    id="boards-button"
    onClick={[Function]}
  >
    <Memo(ForwardRef(ArrowDropDownIcon)) />
    <FormattedMessage
      defaultMessage="Boards"
      id="cboard.components.CommunicatorToolbar.boards"
      values={Object {}}
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Menu))
    anchorEl="someElement"
    className="CommunicatorToolbar__menu"
    id="boards-menu"
    onClose={[Function]}
    open={true}
  >
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-1"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-1-name-key"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
    <WithStyles(ForwardRef(ListItem))
      className="CommunicatorToolbar__menuitem"
      key="board-2"
      onClick={[Function]}
    >
      <WithStyles(ForwardRef(ListItemAvatar))>
        <WithStyles(ForwardRef(Avatar))>
          <Memo(ForwardRef(ViewModuleIcon)) />
        </WithStyles(ForwardRef(Avatar))>
      </WithStyles(ForwardRef(ListItemAvatar))>
      <WithStyles(ForwardRef(ListItemText))
        inset={true}
        primary="board-2-name"
        secondary="cboard.components.CommunicatorToolbar.tiles"
      />
    </WithStyles(ForwardRef(ListItem))>
  </WithStyles(ForwardRef(Menu))>
  <WithWidth(WithMobileDialog)
    onClose={[Function]}
    onSubmit={[Function]}
    open={false}
    title={
      <FormattedMessage
        defaultMessage="Edit Communicator Title"
        id="cboard.components.CommunicatorToolbar.editTitle"
        values={Object {}}
      />
    }
  >
    <WithStyles(ForwardRef(TextField))
      autoFocus={true}
      fullWidth={true}
      label={
        <FormattedMessage
          defaultMessage="Communicator Title"
          id="cboard.components.CommunicatorToolbar.communicatorTitle"
          values={Object {}}
        />
      }
      margin="dense"
      onChange={[Function]}
      required={true}
      type="text"
      value="some-communicator-title"
    />
  </WithWidth(WithMobileDialog)>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--start"
  >
    <WithStyles(ForwardRef(Button))
      className="edit__communicator"
      disabled={false}
      onClick={[Function]}
    >
      <Memo(ForwardRef(LayersIcon))
        className="CommunicatorToolbar__group CommunicatorToolbar__group--start--button"
      />
      cboard.components.CommunicatorToolbar.editCommunicator
    </WithStyles(ForwardRef(Button))>
  </div>
  <div
    className="CommunicatorToolbar__group CommunicatorToolbar__group--end"
  >
    <DefaultBoardSelector
      changeDefaultBoard={[Function]}
    />
  </div>
</div>
`;
